cli/azd/cmd/provision.go

```diff
@@ -185,6 +193,14 @@ func (p *provisionAction) Run(ctx context.Context) (*actions.ActionResult, error
 		}, nil
 	}
 
+	if deployResult.SkippedReason == provisioning.DeploymentStateSkipped {
+		return &actions.ActionResult{
+			Message: &actions.ResultMessage{
+				Header: "There are no changes to provision for your application.",
+			},
```

Comment:
I added an rg

<img width="692" alt="image" src="https://github.com/Azure/azure-dev/assets/2163001/d492ecb5-6a2f-4850-ac08-c258b3952dac">

RG was created

<img width="224" alt="image" src="https://github.com/Azure/azure-dev/assets/2163001/ea01a7ac-7acd-4cfe-9904-4666fc2077d7">


And still got this msg:

<img width="612" alt="image" src="https://github.com/Azure/azure-dev/assets/2163001/d6e999d2-0156-4d68-8fe6-608759578118">

Reply:
PR is been blocked by a recording test. The last version hasn't been created yet.  That's why you are getting those issues from the last version built from this PR.

Right now, the only way to test it is by building locally (until I fixed the recording issue)